### PR TITLE
Fleshing out the zap page.

### PR DESCRIPTION
**Pull request by DavidEBest**
* Moved the images into assets/images
* Added images to the zap page.
* Added more description of the Examining Results, Authentication, Other Tools, and More Info sections.

### CODE REVIEW

File: _pages/security/dynamic-scanning.md [hunk 1]
```diff
@@ -5,29 +5,58 @@ parent: Security
 
 [OWASP ZAP](https://www.owasp.org/index.php/OWASP_Zed_Attack_Proxy_Project) is the tool we use for doing automated vulnerability scanning.
 
+<img src="../../assets/images/zap_home.png" alt="zap home screen"/>
```
Comment (afeld) [inline _pages/security/dynamic-scanning.md]: Minor: mind using [Markdown syntax](https://daringfireball.net/projects/markdown/syntax#img)?

File: _pages/security/dynamic-scanning.md [hunk 2]
```diff
@@ -5,29 +5,58 @@ parent: Security
...
+You will need a running application to test, which you will want to be as production-like as possible. The 'Attack' scripts shouldn't do anything destructive, but will cause a spike in requests and errors. **Inform your team and #devops before running it on live (not locally-hosted) sites.**
 
 1. [Install the ZAP desktop app](https://github.com/zaproxy/zaproxy/wiki/Downloads).
 1. In the Quick Start tab, enter the URL of your application.
     * While you will need to run against production for the official ATO, you can also try out ZAP running against a local server.
-        * If you're using Rails on Mac, start it with
-
-            ```bash
-            rails server --binding 0.0.0.0
-            ```
```
Comment (afeld) [inline _pages/security/dynamic-scanning.md]: Mind if we keep this in? It's an unusual way of starting Rails, so want to make sure we have a note about it.

File: _pages/security/dynamic-scanning.md (hunk 1 above)
Comment (afeld) [inline _pages/security/dynamic-scanning.md]: Ditto for the others that don't need anything besides `src` and `alt` attributes.

File: _pages/security/dynamic-scanning.md [hunk 3]
```diff
@@ -5,29 +5,58 @@ parent: Security
...
 1. Open the Alerts tab.
 
 You can now click on each alert to see more details.
 
+### Examining the Results
+
+#### The Spider
+
+<img src="../../assets/images/spider_results.png" alt="spider results"/>
+As configured, the Quick Start Attack scan does not follow links to other domains or subdomains. If your project uses either, you will want to click <img class="inline" src="../../assets/images/zap_options.png" alt="small options icon"/> and update the options to include the domains & subdomains within the scope.
```
Comment (afeld) [inline _pages/security/dynamic-scanning.md]: Maybe include an example, "e.g. if your assets are hosted on S3".

File: _pages/security/dynamic-scanning.md [hunk 4]
```diff
@@ -5,29 +5,58 @@ parent: Security
...
+
+#### Alerts
+
+<img src="../../assets/images/alert_results.png" alt="alert results"/>
+The Alerts pane lists all alerts discovered while scanning the site. The red and orange-flagged alerts must be taken care of before the application can be ATO'd. You have a little more flexibility when dealing with the yellow and blue flags, but all of them must be either corrected or, in the case of false positives, documented.
```
Comment (afeld) [inline _pages/security/dynamic-scanning.md]: Let's link to https://pages.18f.gov/before-you-ship/security/scanning/#alerts, too.

File: _pages/security/dynamic-scanning.md [hunk 5]
```diff
@@ -5,29 +5,58 @@ parent: Security
...
+
+Some of the alerts are caused by running the application with the debugging web server instead of the approved production server.
```
Comment (afeld) [inline _pages/security/dynamic-scanning.md]: I'd rephrase this to say "If you're running the Attack against a local server, you may see some alerts that you wouldn't see on cloud.gov."

File: _pages/security/dynamic-scanning.md [hunk 6]
```diff
@@ -5,29 +5,58 @@ parent: Security
...
+
 ### Authentication
 
+Testing the application in an authenticated state is easiest when done with ZAP acting as a Man In The Middle Proxy. Setting up with Firefox via the "Plug-n-Hack" extension is available right from the Quick Start page.
```
Comment (afeld) [inline _pages/security/dynamic-scanning.md]: This wasn't working the last time I tried...is it ok now?

Comment (DavidEBest) [inline _pages/security/dynamic-scanning.md]: It worked for me. What part didn't work? The plug-n-hack button, seeing the traffic, or something else?

Comment (afeld) [inline _pages/security/dynamic-scanning.md]: https://github.com/zaproxy/zaproxy/issues/2069

Comment (DavidEBest) [inline _pages/security/dynamic-scanning.md]: Hmm. I've got it working here and I don't remember doing anything special for that. Let me confirm, though.

Comment (DavidEBest) [inline _pages/security/dynamic-scanning.md]: Okay, uninstalling and attempting to re-install didn't work. :/ I'll adjust the recommendation. Good catch.

File: _pages/security/dynamic-scanning.md (hunk 2 above)
Comment (DavidEBest) [inline _pages/security/dynamic-scanning.md]: Is there a reason you need to start a rails app differently? I ran it against a Django app & a Jekyll site, and those just worked.

Comment (afeld) [inline _pages/security/dynamic-scanning.md]: So weird, I was having a problem with it before, but couldn't reproduce with a fresh Rails app or C2.

File: _pages/security/dynamic-scanning.md [hunk 7]
```diff
@@ -5,29 +5,57 @@ parent: Security
...
 
+![zap home screen](../../assets/images/zap_home.png)
+
 ## Usage
 
-You will need a running application to test, which you will want to be as production-like as possible. Before testing, make sure to let your team know, as they may see a spike in requests and errors.
+You will need a running application to test, which you will want to be as production-like as possible. The 'Attack' scripts shouldn't do anything destructive, but will cause a spike in requests and errors. **Inform your team and #devops before running it on live (not locally-hosted) sites.**
 
-1. [Install the ZAP desktop app](https://github.com/zaproxy/zaproxy/wiki/Downloads).
+1. [Install the ZAP desktop app](https://github.com/zaproxy/zaproxy/wiki/Downloads). A recent version of [Java](http://www.oracle.com/technetwork/java/javase/downloads/index.html) (7 or 8 work) is needed to run ZAP.
```
Comment (afeld) [inline _pages/security/dynamic-scanning.md]: Can we remove this new sentence? We should defer to the canonical documentation for any dependencies, and "the Mac OS version includes Java 7" anyway.

Comment (DavidEBest) [inline _pages/security/dynamic-scanning.md]: :+1:

File: _pages/security/dynamic-scanning.md [hunk 8]
```diff
@@ -5,29 +5,57 @@ parent: Security
...
+#### Alerts
+
+![alert results](../../assets/images/alert_results.png)
+The Alerts pane lists all alerts discovered while scanning the site. As described on the [alerts page](../scanning#alerts), the red and orange-flagged alerts must be taken care of before the application can be ATO'd. You have a little more flexibility when dealing with the yellow and blue flags, but all of them must be either corrected or, in the case of false positives, documented.
+
+If you're running the attack against a local server you may see some alerts that you wouldn't see on cloud.gov. Debugging web servers are more 'chatty' about errors than production servers.
+
 ### Authentication
 
+Testing the application in an authenticated state is easiest when done with ZAP acting as a Man In The Middle Proxy. Unfortunately, the "Plug-n-Hack" extension mentioned on the Quick Start page is currently non-functional. By default, ZAP puts an automatic proxy configuration file at http://localhost:8080/proxy.pac. You should set your browser to use that as the proxy for the duration of the testing.
```
Comment (afeld) [inline _pages/security/dynamic-scanning.md]: Can we just link to https://github.com/zaproxy/zap-core-help/wiki/HelpStartProxies instead?

File: _pages/security/dynamic-scanning.md [hunk 9]
```diff
@@ -5,29 +5,57 @@ parent: Security
...
+Authentication can be handled in several ways; the following links provide guidance on handling the most common forms. If these are insufficient, the authentication method is scriptable.
+ 
 * <https://github.com/zaproxy/zap-core-help/wiki/HelpStartConceptsAuthentication>
 * <https://github.com/zaproxy/zaproxy/wiki/FAQformauth>
+
+### Other Tools Within ZAP
+
+#### Ajax Spider
+
+The Attack Script runs the Active Scan and Spider functions. If your page is a javascript-driven single page application, you will want to run the Ajax Spider in addition to the standard spider. The Ajax Spider will instantiate a browser window and spider the application visually. This is significantly slower than the standard spider.
```
Comment (afeld) [inline _pages/security/dynamic-scanning.md]: Mind linking to https://github.com/zaproxy/zap-core-help/wiki/HelpAddonsSpiderAjaxConcepts?

File: _pages/security/dynamic-scanning.md (hunk 8 above)
Comment (DavidEBest) [inline _pages/security/dynamic-scanning.md]: :+1: Good find! Glad I write that process myself.

**Comment by afeld** [comment]
Is this ready for a final review?

**Comment by DavidEBest** [comment]
Yup, I think so.

**Comment by DavidEBest** [comment]
@afeld Ready for re-review.

**Pull request merged**